### PR TITLE
Add API documentation

### PR DESCRIPTION
**Pull request by shmuelk**
## Description

Added API documentation and the ability to generate the HTML documentation using jazzy.
## Motivation and Context

Make Kitura more consumable.
## How Has This Been Tested?

Ran swift build and jazzy.
## Checklist:
- [ ] I have submitted a [CLA form](https://github.com/IBM-Swift/CLA)
- [ ] If applicable, I have updated the documentation accordingly.
- [ ] If applicable, I have added tests to cover my changes.

### CODE REVIEW

File: .jazzy.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+theme: fullwidth
+clean: true
+exclude: [Packages]
+
+readme: README.md
+
+skip_undocumented: false
+hide_documentation_coverage: false
+
+xcodebuild_arguments: [-project, Kitura-TemplateEngine.xcodeproj, -target, KituraTemplateEngine, -toolchain, org.swift.3020160912a, LIBRARY_SEARCH_PATHS=.build/debug]
```
Comment (ianpartridge) [inline .jazzy.yaml]: Now that you can use Swift 3.0 to run Jazzy, this should be:

`xcodebuild_arguments: [-project, Kitura-TemplateEngine.xcodeproj, -target, KituraTemplateEngine, LIBRARY_SEARCH_PATHS=.build/debug]`

File: README.md [hunk 2]
```diff
@@ -1,4 +1,5 @@
 # Kitura-TemplateEngine
-Kitura Template Engine protocol 
+The Kitura Template Engine protocol. Implemented by Templating Engines in order to
+integrate with Kitra's content generation APIs.
```
Comment (ianpartridge) [inline README.md]: Kitura

**Review by ianpartridge — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (shmuelk) [inline README.md]: Fixed here and in the source code as well.

File: .jazzy.yaml (hunk 1 above)
Comment (shmuelk) [inline .jazzy.yaml]: Fixed.

File: README.md (hunk 2 above)
Comment (shmuelk) [reply README.md]: Fixed here and in the source code as well.

File: .jazzy.yaml (hunk 1 above)
Comment (shmuelk) [reply .jazzy.yaml]: Fixed.

**Review by shmuelk — COMMENTED** [review]
Fixed issues.

**Review by ianpartridge — APPROVED** [review]

**Pull request merged**